src/components/pages/BookDetailsPage/BookDetails.tsx

```diff
@@ -0,0 +1,74 @@
+import { Box } from "@mui/material";
+import { Book } from "../../molecules/BookCard/BookCard";
+import { FooterComponent } from "../../organisam/Footer/Footer";
+import { HeaderComponent } from "../../organisam/Header/Header";
+import { RootTemplate } from "../../templates/Template";
+import { BookDetailsComponenet } from "../../organisam/BookDetails/BookDetails";
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+export interface BookDetailsPageComponentProps {}
+
+
+export const BookDetailsPageComponent = (
+  props: BookDetailsPageComponentProps
+) => {
+  let url = window.location.href;
+  let id = Number(url.slice(34));
+  console.log(url);
+
+  console.log(id);
+  const [bookData, setBookData] = useState<Book>();
+  const [dataModifiedToggle,setDataModifiedToggle] = useState(false);
+
+  useEffect(() => {
+    axios
+      .get(`http://localhost:3000/dataBase/${id}`)
+      .then((res: any) => {
+        setBookData(res.data);
+      })
+      .catch((error: any) => console.log(error));
```

Comment:
try to move the axios calls to separate folder API, call the appropriate API whenever needed